docs/content/envoy-primer.md

```diff
@@ -0,0 +1,347 @@
+---
+title: Policy Primer via Examples
+kind: envoy
+weight: 2
+---
+
+This page covers how to write policies for the content of the requests that are passed to OPA by Envoy's
+[External Authorization
+filter](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ext_authz_filter.html).
+
+## Writing Policies
+
+Let's start with an example policy that restricts access to an endpoint based on a user's role and permissions.
+
+```live:bool_example:module:read_only
+package envoy.authz
+
+import input.attributes.request.http as http_request
+
+default allow = false
+
+token = {"valid": valid, "payload": payload} {
+    [_, encoded] := split(http_request.headers.authorization, " ")
+    [valid, _, payload] := io.jwt.decode_verify(encoded, {"secret": "secret"})
+}
+
+allow {
+    is_token_valid
+    action_allowed
+}
+
+is_token_valid {
+  token.valid
+  now := time.now_ns() / 1000000000
+  token.payload.nbf <= now
+  now < token.payload.exp
+}
+
+action_allowed {
+  http_request.method == "GET"
+  token.payload.role == "guest"
+  glob.match("/people*", [], http_request.path)
```

Comment:
The use of `glob.match` in this example isn't quite correct. The default delimiter if omitted is `.`. Are we trying to say:

```
glob.match("/people/*", ["/"], http_request.path)
```

Reply:
Updated